olive/model/handler/pytorch.py

```diff
@@ -284,6 +294,56 @@ def io_config(self) -> Dict[str, Any]:
 
         return io_config
 
+    def merge_kv_cache_hook(self, dummy_inputs, past_kv_names: str = "past_key_values"):
+        if not isinstance(dummy_inputs, dict):
+            return dummy_inputs
+        dummy_inputs = self.past_key_values_input_filter_hook(dummy_inputs)
+        io_config = IoConfig.parse_obj(self.io_config)
+        unused_keys = set()
+        if io_config.kv_cache_config and not dummy_inputs.get(past_kv_names):
+            torch_past_key_values = []
+            k_inputs = io_config.kv_cache_config.get_ort_past_key_names()
+            v_inputs = io_config.kv_cache_config.get_ort_past_value_names()
+            for k_input, v_input in zip(k_inputs, v_inputs):
+                if k_input not in dummy_inputs or v_input not in dummy_inputs:
+                    raise ValueError(f"Cannot find past key-value pair for {k_input} and {v_input} in dummy inputs.")
+                torch_past_key_values.append((dummy_inputs[k_input], dummy_inputs[v_input]))
+                unused_keys.add(k_input)
+                unused_keys.add(v_input)
+            dummy_inputs[past_kv_names] = torch_past_key_values
+        if unused_keys:
+            logger.debug("Merged kv inputs: %s to list for torch model inference", unused_keys)
+            dummy_inputs = exclude_keys(dummy_inputs, unused_keys)
+        return dummy_inputs
+
+    def past_key_values_input_filter_hook(self, dummy_inputs, past_kv_names: str = "past_key_values"):
+        if not isinstance(dummy_inputs, dict):
+            return dummy_inputs
+        # this can happen when we are using an hf dataset to generate dummy inputs
+        # only handle dict for now since we cannot get the name of the input from a list/tuple
+        io_config = self.io_config
+        dummy_input_keys = set(dummy_inputs.keys())
+
+        # after the expansion, user should provide the correct input names for inference
+        for name, dm_input in dummy_inputs.items():
+            # the `past_key_values` is the argument name from huggingface model class
+            # which is independent of the kv-related variables in input list provided by users
+            # if user provided the kv-related variables, we should not remove
+            # the `past_key_values` from dummy inputs. But if not, we should remove it.
+            if (
+                name == past_kv_names
+                and isinstance(dm_input, list)
+                and is_kv_cache_required(dm_input, IoConfig.parse_obj(io_config))
+            ):
+                dummy_input_keys.discard(name)
+
+        unused_keys = dummy_input_keys - set(io_config.get("input_names"))
+
+        if unused_keys:
+            logger.debug("Removing unused dummy inputs: %s", unused_keys)
+            dummy_inputs = exclude_keys(dummy_inputs, unused_keys)
+        return dummy_inputs
+
```

Comment:
seems could be moved to dummy_input and io_config mixin?

Reply:
the two method here seems only depends on io_config dict, which means the new logic could be easily moved to other utils or mixin specific to pytorch kv_cache. Not sure these logic could be reused by onnx model

Reply:
These 2 functions cannot be applied by Onnx model. Here is the though I added the hook function for dummy inputs generation:

Different models(pytorch, onnx) have different inputs which may reqire minor changes to adjust dummy input to feed into different models.

Reply:
could we move them to something like KvCacheMixin/PyKvCacheMixin since the two method only related to kv_cache so we can better organize the code? Originally, the PyTorch model only contains model loading stuff and mixin overrides method

Reply:
Updated.